docs/src/man/comparisons.md

```diff
@@ -41,6 +41,46 @@ Operations|dplyr| DataFrames.jl       |
 |DataFrame as output|`summarize(df, tibble(value = min(x), max(x)))`|`combine(:x => x -> (value = [minimum(x), maximum(x)]), df)`|
 
 
+
+## Comparison with the R package data.table
+
+The following table compares the main functions of DataFrames.jl with the R package data.table (version 1.13):
+
+|Operations              | data.table  | DataFrames.jl|  
+|:-----------------------|:------------|:-------------|
+|Reduce multiple values  |`df[, list(mean(x))]`       |`combine(df, :x => mean)`|
+|Add new columns         |`df[, x_mean := mean(x) ]`  |`transform(df, :x => mean => :x_mean)`|
+|Rename columns          |`setnames(df, "x", "x_new")`|`rename(df, :x => :x_new)`|
+|Pick columns            |`df[, list(x, y)]`          |`select(df, :x, :y)`|
+|Pick & transform columns|`df[, list(mean(x), y)]`    |`select(df, :x => mean, :y)`|
+|Pick rows               |`df[ x >= 1 ]`              |`filter(:x => >=(1), df)`|
+|Sort rows               |`setorder(df, x)`           |`sort(df, :x)`|
+
+As in data.table, some of these functions can be applied by groups:
+
+|Operations              | data.table                                      | DataFrames.jl     |
+|:-----------------------|:------------------------------------------------|:------------------|
+|Reduce multiple values  |`df[, list(mean(x)), by = list(id) ]`            |`combine(grouby(df, :id), :x => mean)`|
+|Add new columns         |`df[, x_mean := mean(x), by = list(id) ]`        |`transform(grouby(df, :id), :x => mean)`|
```

Comment:
```suggestion
|Add new columns         |`df[, x_mean := mean(x), by = list(id) ]`        |`transform!(grouby(df, :id), :x => mean)`|
```

Reply:
Thanks a lot. 
I am dipping my toes into DataFames for a project where we hope to only use one language. So I actually figured out the change in place using `!` in the last couple of days. This is great.

I am not sure what you are referring to regarding the semantics in data.table?
 I might add things such as selection based on strings: `df[, c("x", "y"), with=FALSE ]` which are a little closer to julia. 

The one feature I miss from data.table is to be able to change in place specific rows such as:
```R
df[ x %in% Set, y := newvalue ] 
```
I know its technically doable in DataFrames but I think its a little tricky using just `filter` and `transform`. In any case if we could have something like this I would find it super valuable.

Reply:
I just meant that it would be useful to explain that in DataFrames.jl, one needs to use the functions ending in `!` to modify the data frame in-place, and that indexing using `df[...]` will never modify `df`, while in data.table `:=` does that.

I think `df[ x %in% Set, y := newvalue ]` can be written as `df[in(Set).(df.x), :y] .= newvalue` or `df[in.(df.x, Ref(Set)), :y] .= newvalue` in Julia. Right?

Reply:
Got it. I will add a preamble specifying how `:=` in data.table stands for a change in place and generally corresponds to `!` in julia. 

For the second one, I mostly agree. The `i` in the data.table framework of `df[i, j]` is a little more powerful as it allows merges and more general filtering (though again I am not all that familiar with the DataFrames syntax). 
But for now I will add this example and perhaps raise an issue when a more precise example where the julia idiom is not obvious shows up to me.